src/ORM/DB.php

```diff
@@ -430,6 +430,69 @@ public static function inline_parameters($sql, $parameters)
         return $joined;
     }
 
+
+    /**
+     * Replace 1 parameter with the given value
+     * For example, you can use this to replace a parameter with a column name instead of a literal
+     * @param string $sql The parameterised query
+     * @param int $paramIdx The 0-based position of the parameter
+     * @param array $replacement The value to insert into the queyr
+     * @param bool $skipEscaping Set to true to insert the value as-is, with no escaping. Use with caution!
+     *
+     * @return string
+     */
+    public static function replace_parameter($sql, $paramIdx, $replacement, $skipEscaping = false)
+    {
+        $segments = preg_split('/\?/', $sql);
```

Comment:
`explode` would be faster here.

Reply:
Addressed on parent PR by https://github.com/silverstripe/silverstripe-framework/pull/8380/commits/07f9cdc262769a0e8b9689203c645aab42d6e6b7